### PR TITLE
Rare Crash Fix

### PR DESCRIPTION
**Pull request by osbornjd**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

This PR fixes a few rare crashes that Chris found well exercising the new MDC hardware with the tracking.

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

File: offline/packages/trackreco/PHTpcTrackSeedCircleFit.cc [hunk 1]
```diff
@@ -350,13 +348,16 @@ int PHTpcTrackSeedCircleFit::process_event(PHCompositeNode*)
...
       double R, X0, Y0;
-      CircleFitByTaubin(cpoints, R, X0, Y0);
+      CircleFitByTaubin(globalClusterPositions, R, X0, Y0);
       if(Verbosity() > 2) 
       { std::cout << "PHTpcTrackSeedCircleFit::process_event - track: " << track_key << " R=" << R << " X0=" << X0 << " Y0=" << Y0 << std::endl; }
 
       // set the track x and y positions to the circle PCA
       double dcax, dcay;
       findRoot(R, X0, Y0, dcax, dcay);
+      if(std::isnan(dcax) or std::isnan(dcay))
```
Comment (hupereir) [inline offline/packages/trackreco/PHTpcTrackSeedCircleFit.cc:358]: Hi Joe, this reminds me of the other commit about nan in the Magnetic field lookup. 
Maybe the reason that ::findRoot fails is the R, x0 and y0 are already nan. (as in the other commit), due to the CircleFit actually failing. Can you double check ? 
In this case it should be enough to test on isnan(R) and stop there already. (spares you some extra calculations)

Hugo

Comment (osbornjd) [reply offline/packages/trackreco/PHTpcTrackSeedCircleFit.cc:358]: I did check this, in the case of the crash that Chris found R X0 and Y0 were actually real valued. It turned out that in the case of this particular track X0 was very small (<1e-7) and Y0 was large, so basically R==Y0 and the subtraction returned a negative value which when sqrt-ed gave the nan. So the circle fit succeeded, it was just some very strange edge case where X0<<<Y0

Comment (hupereir) [reply offline/packages/trackreco/PHTpcTrackSeedCircleFit.cc:358]: > I did check this, in the case of the crash that Chris found R X0 and Y0 were actually real valued. It turned out that in the case of this particular track X0 was very small (<1e-7) and Y0 was large, so basically R==Y0 and the subtraction returned a negative value which when sqrt-ed gave the nan. So the circle fit succeeded, it was just some very strange edge case where X0<<<Y0

Ok. Sorry for the noise !
Hugo

Comment (osbornjd) [reply offline/packages/trackreco/PHTpcTrackSeedCircleFit.cc:358]: No worries, thanks for the double check!

**Comment by pinkenburg** [comment]
we have a valgrind complaint in our recent PRs. It does look like the boost header issue we had in spring, cling checking an invalid pointer for no good reason. It seems related to the inclusion of dd4hep into our acts version. I'll have a look later and if it turns out to be bogus (chances are it will be) I'll suppress it

**Pull request merged**